src/triage.rs

```diff
@@ -0,0 +1,138 @@
+use crate::handlers::Context;
+use chrono::{Duration, Utc};
+use hyper::{Body, Response, StatusCode};
+use serde::Serialize;
+use serde_json::value::{to_value, Value};
+use std::sync::Arc;
+use url::Url;
+
+const YELLOW_DAYS: i64 = 7;
+const RED_DAYS: i64 = 14;
+
+pub fn index() -> Result<Response<Body>, hyper::Error> {
+    Ok(Response::builder()
+        .header("Content-Type", "text/html")
+        .status(StatusCode::OK)
+        .body(Body::from(include_str!("../templates/triage/index.html")))
+        .unwrap())
+}
+
+pub async fn pulls(
+    ctx: Arc<Context>,
+    owner: &str,
+    repo: &str,
+) -> Result<Response<Body>, hyper::Error> {
+    let octocrab = &ctx.octocrab;
+    let res = octocrab
+        .pulls(owner, repo)
+        .list()
+        .sort(octocrab::params::pulls::Sort::Updated)
+        .direction(octocrab::params::Direction::Ascending)
+        .per_page(100)
+        .send()
+        .await;
+    let mut page = match res {
+        Ok(page) => page,
+        Err(_) => {
+            return Ok(Response::builder()
+                .status(StatusCode::NOT_FOUND)
+                .body(Body::from("The repository is not found."))
+                .unwrap());
+        }
+    };
+    let mut base_pulls = page.take_items();
+    let mut next_page = page.next;
+    while let Some(mut page) = octocrab
+        .get_page::<octocrab::models::PullRequest>(&next_page)
+        .await
+        .unwrap()
+    {
+        base_pulls.extend(page.take_items());
+        next_page = page.next;
+    }
+
+    let mut pulls: Vec<Value> = Vec::new();
+    for base_pull in base_pulls.into_iter() {
+        let assignee = base_pull.assignee.map_or("".to_string(), |v| v.login);
+        let updated_at = base_pull
+            .updated_at
+            .map_or("".to_string(), |v| v.to_rfc2822());
+
+        let yellow_line = Utc::now() - Duration::days(YELLOW_DAYS);
+        let red_line = Utc::now() - Duration::days(RED_DAYS);
+        let need_triage = match base_pull.updated_at {
+            Some(updated_at) if updated_at <= red_line => "red".to_string(),
+            Some(updated_at) if updated_at <= yellow_line => "yellow".to_string(),
+            _ => "green".to_string(),
+        };
+        let days_from_last_updated_at = if let Some(updated_at) = base_pull.updated_at {
+            (Utc::now() - updated_at).num_days()
+        } else {
+            (Utc::now() - base_pull.created_at).num_days()
+        };
+
+        let labels: String = if let Some(labels) = base_pull.labels {
+            if !labels.is_empty() {
+                labels
+                    .iter()
+                    .map(|label| label.name.clone())
+                    .collect::<Vec<_>>()
+                    .join(",")
+            } else {
+                "".to_string()
+            }
+        } else {
+            "".to_string()
+        };
```

Comment:
Also it could be written:
```suggestion
        let labels = labels.map_or("".to_string(), |labels| {
            labels
                .iter()
                .map(|label| label.name.clone())
                .collect::<Vec<_>>()
                .join(", ")
        });
```